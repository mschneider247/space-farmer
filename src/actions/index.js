export const setUser = user => ({
  type: 'SET_USER_NAME',
  user
});

export const setProposal = proposal => ({
  type: 'SET_PROPOSAL',
  proposal
})

export const addProposal = proposal => ({
  type: 'ADD_PROPOSAL',
  proposal
})