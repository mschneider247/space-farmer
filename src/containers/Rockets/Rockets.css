#where_prompt {
  margin-left: 8%;
}

.destinations, .rockets {
  width: 100%;
  margin-left: 3%;
  display: flex;
  flex-wrap: wrap;
  border-bottom: 3px solid grey;
}

.destination_image {
  width: 140px;
  height: 140px;
  border-radius: 10px;
}

.destination-card, .rocket-card {
  border: 1px solid grey;
  border-radius: 10px;
  margin: 1%;
  text-align: center;
}

.destination-card {
  margin-top: 10%;
  width: 160px;
  height: 200px;
}

.destination_p {
  height: 20px;
}

.rocket-card {
  width: 220px;
}

.rocket_image {
  height: 420px;
}

.destination_input, .rocket_input {
  margin-left: 45%;
}

.rocket-footer_h3 {
  color: black;
}

.rocket-footer {
  width: 30%;
  padding: 1%;
  position: fixed;
  top: 14%;
  right: 2%;
  margin-left: 12%;
  display: flex;
  flex-direction: column;
  background-color: grey;
  border: 1px solid lime;
  border-radius: 10px;
  text-align: center;
}