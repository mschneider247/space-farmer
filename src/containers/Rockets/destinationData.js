import LEO from '../../images/LEO.jpg';
import GTO from '../../images/GTO.jpg';
import moon from '../../images/moon.jpg';
import mars from '../../images/mars.jpg';
import pluto from '../../images/pluto.png';

const destinationData = [
  {
    id: 1,
    destination: 'leo',
    text: 'Low Earth Orbit',
    imageURL: LEO,
    isChosen: false,
    distance: 600,
  },
  {
    id: 2,
    destination: 'gto',
    text: 'Geosynchronous Transfer Orbit',
    imageURL: GTO,
    isChosen: false,
    distance: 22236,
  },
  {
    id: 3,
    destination: 'moon',
    text: 'Moon',
    imageURL: moon,
    isChosen: false,
    distance: 238900,
  },
  {
    id: 4,
    destination: 'mars',
    text: 'Mars',
    imageURL: mars,
    isChosen: false,
    distance: 140000000,
  },
  {
    id: 5,
    destination: 'pluto',
    text: 'Pluto',
    imageURL: pluto,
    isChosen: false,
    distance: 3648520000
  }
];

export default destinationData;