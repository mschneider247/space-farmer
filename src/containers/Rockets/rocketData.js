import falcon1 from '../../images/falcon1.jpg';
import falcon9 from '../../images/falcon9.jpg';
import falconHeavy from '../../images/falconHeavy.jpg';
import starship from '../../images/starship.jpg';

const rocketData = [
  {
    id: 1,
    name: 'Falcon 1',
    cost: 6700000,
    payloads: [
      {
        id: "leo",
        name: "Low Earth Orbit",
        lb: 992, 
      },
      {
        id: "gto",
        name: "Geosynchronous Transfer Orbit",
        lb: 361,
      },
    ],
    image: falcon1,
    isChosen: false,
  },
  {
    id: 2,
    name: 'Falcon 9',
    cost: 62000000,
    payloads: [
      {
        id: "leo",
        name: "Low Earth Orbit",
        lb: 50265,
      },
      {
        id: "gto",
        name: "Geosynchronous Transfer Orbit",
        lb: 18300,
      },
      {
        id: "moon",
        name: "Moon",
        lb: 13572,
      },
      {
        id: "mars",
        name: "Mars",
        lb: 8860,
      },
    ],
    image: falcon9,
    isChosen: false,
  },
  {
    id: 3,
    name: 'Falcon Heavy',
    cost: 90000000,
    payloads: [
      {
        id: "leo",
        name: "Low Earth Orbit",
        lb: 140660,
      },
      {
        id: "gto",
        name: "Geosynchronous Transfer Orbit",
        lb: 58860,
      },
      {
        id: "moon",
        name: "Moon",
        lb: 37978,
      },
      {
        id: "mars",
        name: "Mars",
        lb: 37040,
      },
      {
        id: "pluto",
        name: "Pluto",
        lb: 7720,
      },
    ],
    image: falconHeavy,
    isChosen: false,
  },
  {
    id: 4,
    name: 'Starship',
    cost: 7000000,
    payloads: [
      {
        id: "leo",
        name: "Low Earth Orbit",
        lb: 330000,
      },
      {
        id: "gto",
        name: "Geosynchronous Transfer Orbit",
        lb: 291720,
      },
      {
        id: "moon",
        name: "Moon",
        lb: 220000,
      },
      {
        id: "mars",
        name: "Mars",
        lb: 220000,
      },
      {
        id: "pluto",
        name: "Pluto",
        lb: 24090,
      },
    ],
    image: starship,
    isChosen: false,
  },
];

export default rocketData;