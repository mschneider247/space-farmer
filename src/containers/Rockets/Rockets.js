import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'
import { connect } from 'react-redux';
import { addRocketChoice, addDestination } from '../../actions';
import './Rockets.css';
import LEO from '../../images/LEO.jpg';
import GTO from '../../images/GTO.jpg';
import moon from '../../images/moon.jpg';
import mars from '../../images/mars.jpg';
import pluto from '../../images/pluto.png';

export class Rockets extends Component {
  constructor() {
    super();
    this.state = {
      destination: '',
      rocketChoice: '',
    };
  }

  handleDestination = (destination) => {
    this.setState({ destination: destination })
    this.props.addDestination(destination)
  }

  handleRocket = (rocket) => {
    this.setState({ rocketChoice: rocket.name });
    this.props.addRocketChoice(rocket);
  }

  createRockets = (availableRockets) => {
    return availableRockets.map(rocket => {
      return (
        <div className="rocket-card" key={rocket.id}>
          <img className="rocket_image" src={rocket.image} alt={rocket.name}/>
          <p>{rocket.name}</p>
          <p>${rocket.cost}</p>
          <input className="rocket_input" onClick={() =>{this.handleRocket(rocket)}} type="radio" name="chooseRocket" value={rocket.name}/>
        </div>
      )
    })
  }

  getAvailableRockets = () => {
    let currentDestination = this.props.proposals[this.props.proposals.length - 1].destination;
    let availableRockets = [];
    this.props.rockets.forEach(rocket => {
      rocket.payloads.forEach(payload => {
        if (payload.id === currentDestination){
          availableRockets.push(rocket);
        }
      })
    })
    let mappedRockets = this.createRockets(availableRockets);
    return mappedRockets
  }

  render() {
    return (
      <>
        <h3>Where do you want to go?</h3>
        <section className="destinations">
           <div className="destination-card">
            <p className="destination_p">Low Earth Orbit</p>
            <img src={LEO} alt="LEO" />
            <input className="destination_input" onClick={() =>{ this.handleDestination("leo")}} type="radio" name="destination" value="leo" />
          </div>
          <div className="destination-card">
            <p className="destination_p">Geostationary Transfer Orbit</p>
            <img src={GTO} alt="GTO" />
            <input className="destination_input" onClick={() =>{ this.handleDestination("gto")}} type="radio" name="destination" value="gto" />
          </div>
          <div className="destination-card">
            <p className="destination_p">Moon</p>
            <img src={moon} alt="Moon" />
            <input className="destination_input" onClick={() =>{ this.handleDestination("moon")}} type="radio" name="destination" value="moon" />
          </div>
          <div className="destination-card">
            <p className="destination_p">Mars</p>
            <img src={mars} alt="Mars" />
            <input className="destination_input" onClick={() =>{ this.handleDestination("mars")}} type="radio" name="destination" value="mars" />
          </div>
          <div className="destination-card">
            <p className="destination_p">Pluto</p>
            <img src={pluto} alt="Pluto" />
            <input className="destination_input" onClick={() =>{ this.handleDestination("pluto")}} type="radio" name="destination" value="pluto" />
          </div>
        </section>
        <section className="rockets">
          {this.props.proposals[this.props.proposals.length -1].destination ? 
            this.getAvailableRockets()
            : null
          }
        </section>
        <section className="rocket-footer">
          <h3 className="rocket-footer_h3">Destination: 
            {this.state.destination}
          </h3>
          <h3 className = "rocket-footer_h3"> Rocket Choice: 
            {this.state.rocketChoice}
          </h3>
          {(this.state.destination !== '' && this.state.rocketChoice !== '') ? <Link to='/overview'><button >Continue To Overview</button></Link> : null}
        </section>
      </>
    )
  }
};

export const mapStateToProps = (state) => ({
  proposals: state.proposals,
  rockets: state.rockets,
});

export const mapDispatchToProps = (dispatch) => ({
  addDestination: (destination) => dispatch(addDestination(destination)),
  addRocketChoice: (rocket) => dispatch(addRocketChoice(rocket))
});

export default connect(mapStateToProps, mapDispatchToProps)(Rockets);

Rockets.propTypes = {
  proposals: PropTypes.array,
  rockets: PropTypes.array,
  addDestination: PropTypes.func,
  addRocketChoice: PropTypes.func,
  destination: PropTypes.string,
  rocketChoice: PropTypes.string
}