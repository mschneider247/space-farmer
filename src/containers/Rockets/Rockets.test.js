import React from 'react';
import { shallow } from 'enzyme';
import { Rockets, mapDispatchToProps, mapStateToProps } from './Rockets';
import { addDestination } from '../../actions';

describe('Rockets container', () => {
  const proposals = [{ id: 1 }, { id: 2 }];
  const addDestination = jest.fn();
  const addRocketChoice = jest.fn();
  const wrapper = shallow(<Rockets proposals={proposals} addDestination={addDestination} addRocketChoice={addRocketChoice}/>)

  it('Rockets should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call addDestination when the first input is clicked', () => {
    wrapper.find('input').first().simulate('click');
    expect(addDestination).toHaveBeenCalled();
  });

  it('Should return the correct state when handleDestination is called', () => {
    const expectedState = {
      destination: 'mars',
      rocketChoice: '',
    };
    wrapper.instance().handleDestination('mars');
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('Should return the correct state when handleRocket is called', () => {
    const expectedState = {
      destination: 'mars',
      rocketChoice: 'Big Rocket',
    };
    wrapper.instance().handleRocket({ name: 'Big Rocket' });
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('Should call createRockets to iterate over rockets', () => {
    const returnedValue = wrapper.instance().createRockets([]);
    const expected = [];
    expect(returnedValue).toEqual(expected);
  });
});

describe('Rockets mapDispatchToProps', () => {
  it('Should dispatch destination through addDestination', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = addDestination('moon');
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.addDestination('moon');
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});

describe('Rockets mapStateToProps', () => {
  it('Should access proposals in the Redux store', () => {
    const mockState = {
      proposals: { id: 1 },
    };
    const expected = { proposals: { id: 1 } };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('Should access rockets in the Redux store', () => {
    const mockState = {
      rockets: { name: 'rocket-thing' },
    };
    const expected = { rockets: { name: 'rocket-thing' } };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
