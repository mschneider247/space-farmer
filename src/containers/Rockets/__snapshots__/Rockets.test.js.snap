// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets container Rockets should match its snapshot 1`] = `ShallowWrapper {}`;
